### PR TITLE
[SDK TypeBinding] Improve error logging for failed deserialization for POCO scenarios

### PR DESCRIPTION
**Pull request by aishwaryabh**
### Issue describing the changes in this PR
This PR adds specific error logging if deserialization fails since the POCO was expecting a json instead of a string.
resolves https://github.com/Azure/azure-functions-dotnet-worker/issues/1329

### Pull request checklist

* [x] My changes **do not** require documentation changes
  * [ ] Otherwise: Documentation issue linked to PR
* [x] My changes **should not** be added to the release notes for the next release
  * [ ] Otherwise: I've added my notes to `release_notes.md`
* [x] My changes **do not** need to be backported to a previous version
  * [ ] Otherwise: Backport tracked by issue/PR #issue_or_pr
* [x] I have added all required tests (Unit tests, E2E tests)

### Additional information

Additional PR information

### CODE REVIEW

**Review by jviau — APPROVED** [review]

File: test/WorkerExtensionTests/Blob/BlobStorageConverterTests.cs [hunk 1]
```diff
@@ -771,6 +771,40 @@ public async Task ConvertAsync_POCO_InvalidJson_ReturnsFailed()
             Assert.IsType<JsonException>(conversionResult.Error);
         }
 
+        [Fact]
+        public async Task ConvertAsync_ValidModelBindingData_POCOCollection_Array_ReturnsFailed()
```
Comment (kshyju) [inline test/WorkerExtensionTests/Blob/BlobStorageConverterTests.cs:775]: nit: The test is confirming incorrect JSON causes conversion failure. correct? The testname has "ValidModelBindingData". Should that be improved?

**Review by kshyju — APPROVED** [review]

File: extensions/Worker.Extensions.Storage.Blobs/src/BlobStorageConverter.cs [hunk 2]
```diff
@@ -105,6 +107,17 @@ private async ValueTask<ConversionResult> ConvertFromCollectionBindingDataAsync(
 
                 return ConversionResult.Success(result);
             }
+            catch (JsonException ex)
```
Comment (kshyju) [inline extensions/Worker.Extensions.Storage.Blobs/src/BlobStorageConverter.cs:110]: Are we only making this change to Blob extension converter? We have a built-in JSonPocoConverter as well. Wonder we should do similar thing there as well.

**Review by surgupta-msft — APPROVED** [review]

File: extensions/Worker.Extensions.Storage.Blobs/src/BlobStorageConverter.cs (hunk 2 above)
Comment (kshyju) [reply extensions/Worker.Extensions.Storage.Blobs/src/BlobStorageConverter.cs:110]: That probably is a breaking change. So may be something we can discuss with the team and address later in a separate PR.

Comment (aishwaryabh) [reply extensions/Worker.Extensions.Storage.Blobs/src/BlobStorageConverter.cs:110]: Yeah this PR specifically focuses on the use case for blob for POCO serialization. We can address this in another PR

File: test/WorkerExtensionTests/Blob/BlobStorageConverterTests.cs (hunk 1 above)
Comment (aishwaryabh) [reply test/WorkerExtensionTests/Blob/BlobStorageConverterTests.cs]: Changed!

**Pull request merged**